test/integration/offline.js

```diff
@@ -3,67 +3,99 @@
+    context('with an exiting lambda-proxy integration type route', () => {
+        it('should return the expected status code', (done) => {
+            const offLine = new OffLineBuilder().addFunctionConfig('fn1', {
+                handler: 'handler.hello',
+                events: [{
+                    http: {
+                        path: 'fn1',
+                        method: 'GET',
+                    },
+                }],
+            }, (event, context, cb) => {cb(null, {
+                statusCode: 201,
+                body: null,
+            })}).toObject();
 
-      server.inject(reqOpts, (res) => {
-        expect(res.statusCode).to.eq(404);
-      });
+            offLine.inject({
+                method: 'GET',
+                url: '/fn1',
+            }, (res) => {
+                expect(res.statusCode).to.eq(201);
+                done();
+            });
+        });
     });
-  });
 
-  context('with an exiting lambda-proxy integration type route', () => {
-    it('should return the expected status code', () => {
-      const reqOpts = {
-        method: 'GET',
-        url: '/fn1',
-      };
+    context('with an exiting lambda-proxy integration type route [SHORT MODE]', () => {
+        it('should return the expected status code', (done) => {
+            const offLine = new OffLineBuilder().addFunctionHTTP('hello', {
+                        path: 'fn1',
+                        method: 'GET'
+            }, (event, context, cb) => {cb(null, {
+                statusCode: 201,
+                body: null,
+            })}).toObject();
 
-      sinon.stub(functionHelper, 'createHandler', () => () => ({
-        statusCode: 201,
-        body: null,
-      }));
+            offLine.inject({
+                method: 'GET',
+                url: '/fn1',
+            }, (res) => {
+                expect(res.statusCode).to.eq(201);
+                done();
+            });
+        });
+    });
 
-      server.inject(reqOpts, (res) => {
-        expect(res.statusCode).to.eq(201);
-      });
+    context('handling html', () => {
+        it.skip('should handle html end point configuration', (done) => {
+            const offLine = new OffLineBuilder().addFunctionConfig('index', {
+                handler: 'users.index',
+                events: [{
+                    //http: 'GET users/index'
+                    http: {
+                        path: 'index',
+                        method: 'GET',
+                        response: {
+                            headers: {
+                                'Content-Type': "'text/html'"
+                            },
+                            template: "$input.path('$')"
+                        }
+                    }
+                }]
+            }, (event, context, cb) => cb(null, 'Hello World')).toObject();
 
-      functionHelper.createHandler.restore();
+            offLine.inject("/index", (res) => {
+                expect(res.statusCode).to.eq(201);
```

Comment:
Unless you put it differently, for lambda integration type the statusCode is a string and always '200'. So you should change it to '200' instead of 201.